appstore/developer_portal_api.py

```diff
@@ -0,0 +1,558 @@
+            return jsonify(error="Invalid value for field: visible", e="invalid.field.visible"), 400
+
+        # Disallow change face category
+        if "category" in req and app.category == "Faces":
+            return jsonify(error="Cannot change category for watchface", e="disallowed.field.category"), 400
+
+        # Check title length
+        if "title" in req and len(req["title"]) > 45:
+            return jsonify(error="Title must be less than 45 characters", e="invalid.field.title"), 400
+            
+        # Update the app
+        for x in req:
+            setattr(app, x, req[x])
+
+        # Updating description requires iterating through asset collection
+        if "description" in req:
+            for x in app.asset_collections:
+                app.asset_collections[x].description = req["description"]
+
+        db.session.commit()
+        if algolia_index:
+            algolia_index.partial_update_objects([algolia_app(app)], { 'createIfNotExists': False })
+
+        return jsonify(success=True, id=app.id)
+
+
+@devportal_api.route('/app/<app_id>/release', methods=['POST'])
+def submit_new_release(app_id):
+    try:
+        app = App.query.filter(App.id == app_id).one()
+    except NoResultFound:
+        return jsonify(error="Unknown app", e="app.notfound"), 400
+
+    # Check we own the app
+    if not is_users_developer_id(app.developer_id):
+        return jsonify(error="You do not have permission to modify that app", e="permission.denied"), 403
+
+    data = dict(request.form)
+
+    if "pbw" not in request.files:
+        return jsonify(error="Missing file: pbw", e="pbw.missing"), 400
+
+    if "release_notes" not in data:
+        return jsonify(error="Missing field: release_notes", e="release_notes.missing"), 400
+
+    pbw_file = request.files['pbw'].read()
+
+    try:
+        pbw = PBW(pbw_file, 'aplite')
+        with pbw.zip.open('appinfo.json') as f:
+            appinfo = json.load(f)
+    except BadZipFile as e:
+        return jsonify(error=f"Your pbw file is invalid or corrupted", e="invalid.pbw"), 400
+    except KeyError as e:
+        return jsonify(error=f"Your pbw file is invalid or corrupted", e="invalid.pbw"), 400
+
+    appinfo_valid, appinfo_valid_reason = is_valid_appinfo(appinfo)
+    if not appinfo_valid:
+        return jsonify(error=f"The appinfo.json in your pbw file has the following error: {appinfo_valid_reason}", e="invalid.appinfocontent"), 400
+
+    uuid = appinfo['uuid']
+    version = appinfo['versionLabel']
+        
+    if uuid != app.app_uuid:
+        return jsonify(error="The UUID in appinfo.json does not match the app you are trying to update", e="uuid.mismatch"), 400
+
+    release_old = Release.query.filter_by(app=app).order_by(Release.published_date.desc()).first()
+
+    if version <= release_old.version:
+        return jsonify(
+            error=f"The version ({version}) is already on the appstore", 
```

Comment:
`a newer version than {version}`?

Reply:
The version in appinfo.json must be higher than that already on the store when submitting a new release